kinematics_base_test/src/test_kinematics_plugin.cpp

```diff
@@ -483,23 +520,121 @@ TEST(IKFastPlugin, getIKMultipleSolutions)
+      kinematics_test.kinematics_solver_->getJointNames().size(), 0.0);
 
+  std::vector<std::string> fk_names;
+  fk_names.push_back(kinematics_test.kinematics_solver_->getTipFrame());
+  robot_state::RobotState kinematic_state(kinematic_model);
 
+  double min = -3.14;
+  double max = 3.14;
+  double rnd;
 
+  unsigned int success = 0;
+  unsigned int num_ik_solutions = 0;
+  ros::WallTime start_time = ros::WallTime::now();
+  for (unsigned int i = 0; i < kinematics_test.num_nearest_ik_tests_; ++i) {
+    seed.resize(kinematics_test.kinematics_solver_->getJointNames().size(),
+                0.0);
+    fk_values.resize(kinematics_test.kinematics_solver_->getJointNames().size(),
+                     0.0);
+    kinematic_state.setToRandomPositions(joint_model_group);
+    kinematic_state.copyJointGroupPositions(joint_model_group, fk_values);
+    std::vector<geometry_msgs::Pose> poses;
+    poses.resize(1);
+
+    for (std::size_t i = 0; i < seed.size(); i++) {
+      double rnd = (double)rand() / (double)RAND_MAX;
+      seed[i] = min + rnd * (max - min);
+    }
+
+    ASSERT_TRUE(kinematics_test.kinematics_solver_->getPositionFK(
+        fk_names, fk_values, poses));
+
+    // uncomment next line to see the behavior change of KDL plugin
+    //    seed = fk_values;
+
+    // getIK
+    solution_getIK.clear();
+    kinematics_test.kinematics_solver_->getPositionIK(
+        poses[0], seed, solution_getIK, error_code);
+    if (error_code.val == error_code.SUCCESS) {
+      double error_getIK = 0.0;
+      for (std::size_t j = 0; j < seed.size(); ++j) {
+        error_getIK = error_getIK + fabs(solution_getIK[j] - seed[j]);
+      }
+
+      solutions.clear();
+      kinematics_test.kinematics_solver_->getPositionIK(poses, seed, solutions,
+                                                        result, options);
+      ROS_DEBUG("Pose: %f %f %f", poses[0].position.x, poses[0].position.y,
+                poses[0].position.z);
+      ROS_DEBUG("Orient: %f %f %f %f", poses[0].orientation.x,
+                poses[0].orientation.y, poses[0].orientation.z,
+                poses[0].orientation.w);
+
+      if (result.kinematic_error == kinematics::KinematicErrors::OK) {
+        double smallest_error_multipleIK = DBL_MAX;
+        for (unsigned int i = 0; i < solutions.size(); i++) {
+          std::vector<double> &solution_multipleIK = solutions[i];
+          double error_multipleIK = 0.0;
+          for (std::size_t j = 0; j < seed.size(); ++j) {
+            error_multipleIK =
+                error_multipleIK + fabs(solution_multipleIK[j] - seed[j]);
+          }
+
+          if (error_multipleIK <= smallest_error_multipleIK) {
+            smallest_error_multipleIK = error_multipleIK;
+          }
+        }
+        ASSERT_DOUBLE_EQ(smallest_error_multipleIK, error_getIK);
+      } else {
+        ROS_INFO_STREAM("Something is wrong. It shouldn't be happening");
+      }
+    } else {
+      ROS_INFO_STREAM("No Solution Found");
```

Comment:
Should this not be an error or warning? Furthermore, this shouldn't fail as the tool pose was obtained from FK.